src/Microsoft.SqlTools.Hosting/Hosting/Protocol/IProtocolEndpoint.cs

```diff
@@ -18,15 +18,18 @@ public interface IProtocolEndpoint : IEventSender, IRequestSender
     {
         void SetRequestHandler<TParams, TResult>(
             RequestType<TParams, TResult> requestType,
-            Func<TParams, RequestContext<TResult>, Task> requestHandler);
+            Func<TParams, RequestContext<TResult>, Task> requestHandler,
+            bool isParallelProcessingSupported = false);

```

Comment:
Please add doc comments for these methods and at the very least add a param comment for the new param being added (but if you could do the others while you're here that'd be great)